students/jungchan/users/views.py

```diff
@@ -1,19 +1,23 @@
-import json, re
+import json, bcrypt
 
 from django.http import JsonResponse
 from django.views import View
 
 from users.models import User
 from users.validation import valid_email, valid_password
 
+
 class SignupView(View):
     def post(self, request):
         try:
-            data         = json.loads(request.body)
-            name         = data['name']
-            email        = data['email']
-            password     = data['password']
-            phone_number = data['phone_number']
+            data             = json.loads(request.body)
+            encoded_password = data['password'].encode('utf-8')
+            hashed_password  = bcrypt.hashpw(encoded_password, bcrypt.gensalt())
+            decoded_password = hashed_password.decode('utf-8')
+            name             = data['name']
+            email            = data['email']
+            password         = data['password']
+            phone_number     = data['phone_number']
```

Comment:
```suggestion
            data             = json.loads(request.body)
            name             = data['name']
            email            = data['email']
            password         = data['password']
            phone_number     = data['phone_number']
```

- 비밀번호 암호화 하는 부부은 마지막 if 뒤에 오면 좋을 것 같습니다.
- 프론트에서 받은 값들에 대한 검사가 다 끝난 뒤에 우리가 원하는 로직을 실행하는 것이 조금 더 효율적입니다.